docs/madr/decisions/041-meshservice-kubernetes-ux.md

```diff
@@ -0,0 +1,94 @@
+# `MeshService` Kubernetes UX
+
+* Status: accepted
+
+Technical Story: #9704
+
+## Context and Problem Statement
+
+The new `MeshService` resource in Kubernetes zones maps 1-1 to Kubernetes
+`Service` resources. Additionally, users should not have to manage
+`MeshServices` directly but instead a `MeshService` should be derived directly
+from its `Service` with the possibility of users annotating the `Service`
+to control specifics.
+
+## Considered Options
+
+* CP manages `MeshService` from `Service`
+* Users have to create `MeshService`
+
+## Decision Outcome
+
+We will convert `Services` to `MeshServices` in a Kubernetes controller,
+assuming:
+
+1. The namespace is part of the mesh with `kuma.io/sidecar-injection: "true"`
+  or the `Service` is labelled with `kuma.io/mesh`.
```

Comment:
We should consider having the same label for all k8s objects to enable the mesh. Today namespaces and pods need `kuma.io/sidecar-injection` to enable sidecars. It can be surprising for users if the service needs only `kuma.io/mesh` to get sidecars

Reply:
Yeah I agree but `kuma.io/sidecar-injection` fits so badly. @jakubdyszkiewicz @lahabana opinions?

Reply:
> It can be surprising for users if the service needs only kuma.io/mesh to get sidecars

They don't enable sidecars this way. They opt-in a Service to a mesh.

Let's also keep in mind that it's probably 10-20% of use cases. The majority of time people are placing sidecar-injection on the namespace and then disable mesh on specific pods if needed.

Reply:
> They don't enable sidecars this way. They opt-in a Service to a mesh.

I see, so the `kuma.io/mesh` label just means we're going to convert the Service to a MeshService. But what's the usecase for that, why'd you want to introduce a Service to a mesh while all pods implementing the Service don't have a sidecar? Is it something related to ExternalName?

Reply:
The use case is that there needs to be a mechanism to convert Service to MeshService for a corresponding Dataplane object that has sidecar-injection.

We could retrieve Pods in Service Controller to check this, but the controller would have to have a dependency on pods now which we want to avoid.